### PR TITLE
Combobox: Support aria-posinset for input values when dropdown is closed

### LINKED ISSUE #11209: [Accessibility][Usable]: Narrator doesn't narrates the position number of element "Option d" without expanding it inside Basic uncontrolled example (allowFreeform: T, AutoComplete: T).
Split out of issue #6920

### PR DESCRIPTION
**Pull request by micahgodbolt**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #11209
- [ ] Include a change request file using `$ yarn change`

#### Description of changes
Preview of possible solution.

Need to move SizePosCache into shared place, and resolve how to get posinset to announce on each change without repeating the label each time.

#### Focus areas to test

(optional)

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/11511)

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx [hunk 1]
```diff
@@ -408,6 +423,10 @@ export class ComboBox extends BaseComponent<IComboBoxProps, IComboBoxState> {
                 aria-autocomplete={this._getAriaAutoCompleteValue()}
                 role="combobox"
                 readOnly={disabled || !allowFreeform}
+                aria-posinset={this._sizePosCache.positionInSet(
+                  currentPendingValueValidIndex > 0 ? currentPendingValueValidIndex : selectedIndices[0]
```
Comment (jdhuntington) [inline packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx]: Maybe ` currentPendingValueValidIndex >= 0` instead? Looks like `0` is a valid value if I'm reading existing code correctly.

Comment (micahgodbolt) [reply packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx]: Fixed.

**Comment by micahgodbolt** [comment]
Aria-live on an input element seems to have some limits. It appears to only work properly if the input is associated with a label (not sure why). And the label is read out every time the value changes. With a short title, this is fine. With a long title this is very annoying. Fortunately, this is only needed when the dropdown is closed. An open dropdown is not affected.

**Comment by micahgodbolt** [comment]
@cschlechty Is this aria-posinset information worth the label being repeated on each change? I feel like this is 1 step forward, and 1 step back.

**Comment by micahgodbolt** [comment]
Here's the readout of the "Custom Styled Combobox" 

image


> Custom styled ComboBox, Arial Black, Combo Edit, read-only, <say-as interpret-as="spell">1</say-as> of <say-as interpret-as="spell">4</say-as>, 
> Arial Black selected
> Custom styled ComboBox, Times New Roman, read-only, <say-as interpret-as="spell">2</say-as> of <say-as interpret-as="spell">4</say-as>, 
> Times New Roman selected
> Custom styled ComboBox, Comic Sans MS, read-only, <say-as interpret-as="spell">3</say-as> of <say-as interpret-as="spell">4</say-as>, 
> Comic Sans MS selected
> Exiting Narrator

**Comment by micahgodbolt** [comment]
pinged Chris in teams. Would like to know if tradeoff of extra noise is worth posinset being read off.

File: packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx [hunk 2]
```diff
@@ -191,12 +194,16 @@ export class ComboBox extends BaseComponent<IComboBoxProps, IComboBoxState> {
...
+    this._sizePosCache.updateOptions(props.options);
+
     this.state = {
       isOpen: false,
       selectedIndices: initialSelectedIndices,
       focused: false,
       suggestedDisplayValue: undefined,
+      // suggestedDisplayIndex: undefined,
       currentOptions: this.props.options,
+      // currentPendingValueValidIndex: initialSelectedIndices.length ? initialSelectedIndices[0] : -1,
```
[the remark this answers is not in the document]
Comment (micahgodbolt) [reply packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx]: fixed

**Comment by micahgodbolt** [comment]
After chatting with @cschlechty and looking at the `aria-posinset` spec, it does not look like this is the right choice.

The value that we are looking to read out is role="combobox", which `aria-posinset` is not valid on. Adding this attribute would just cause errors in the future. 

Closing this PR down, and looking at other solutions.

**Pull request closed without merging**